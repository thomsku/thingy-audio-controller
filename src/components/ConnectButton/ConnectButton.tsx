import React from "react";

interface IProps {
  connected: boolean;
  connect(): void;
}

const ConnectButton: React.FunctionComponent<IProps> = (props) => {
  return (
    <div className="buttondiv">
      <button className="connectbutton" onClick={props.connect}>
        {props.connected ? (
          "disconnect"
        ) : (
          "connect"
        )}
      </button>
    </div>
  );
};

export default ConnectButton;
