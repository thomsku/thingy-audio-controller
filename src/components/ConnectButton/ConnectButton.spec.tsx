import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import ConnectButton from "./ConnectButton";

describe("ConnectButton", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => wrapper = shallow(<ConnectButton connect={jest.fn()} connected={false}/>));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });
});
