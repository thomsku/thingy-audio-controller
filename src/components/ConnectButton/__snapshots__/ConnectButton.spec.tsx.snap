// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectButton should render correctly 1`] = `ShallowWrapper {}`;
